tests/test_build.py

```diff
@@ -40,6 +40,21 @@ def test_respects_none_overrides() -> None:
     assert result.hobbies is None
 
 
+def test_allows_nested_optional_field_overrides() -> None:
+    class MyChildModel(BaseModel):
+        name: str
+
+    class MyParentModel(BaseModel):
+        child: MyChildModel | None
+
+    class MyParentModelFactory(ModelFactory):
+        __model__ = MyParentModel
+
+    MyParentModelFactory.seed_random(2)
```

Comment:
```suggestion
    class MyParentModelFactory(ModelFactory[MyParentModel]):
        __model__ = MyParentModel

        @classmethod
        def should_set_none_value(cls, field_meta: FieldMeta) -> bool:
            return True
```

This removes any of the randomness.

Reply:
https://github.com/litestar-org/polyfactory/pull/420/commits/315b77bc554ed44c53eda888604bd5b3619ea6c4